src/Config.ts

```diff
@@ -621,17 +627,28 @@ export class Config implements ConfigParams, ParserConfig {
         throw new ConfigValueEmpty('currencySymbol')
       }
     })
-    this.validateNumberToBeAtLeast(this.maxColumns, 'maxColumns', 1)
+    validateNumberToBeAtLeast(this.maxColumns, 'maxColumns', 1)
 
-    this.checkIfParametersNotInConflict(
+    configCheckIfParametersNotInConflict(
       {value: this.decimalSeparator, name: 'decimalSeparator'},
       {value: this.functionArgSeparator, name: 'functionArgSeparator'},
       {value: this.thousandSeparator, name: 'thousandSeparator'}
     )
   }
 
-  public getConfig(): ConfigParams { //TODO: avoid pollution
-    return this
+  public getConfig(): ConfigParams {
+    const ret: Partial<ConfigParams> = {}
+    for (const key in Config.defaultConfig) {
+      const val = this[key as ConfigParamsList]
+      if (Array.isArray(val)) {
+        //eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+        (ret[key as ConfigParamsList] as ConfigParamsTypes) = [ ...val ]
+      } else {
+        //eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+        (ret[key as ConfigParamsList] as ConfigParamsTypes) = val
+      }
+    }
+    return ret as ConfigParams
```

Comment:
😬 Maybe something more generic? Those types are assigning anything to anything anyway.

```suggestion
  public getConfig(): ConfigParams {
    const ret: { [key: string]: any } = {};

    for (let [key, val] of Object.entries(Config.defaultConfig)) {
      if (Array.isArray(val)) {
        ret[key] = [ ...val ]
      } else {
        ret[key] = val
      }
    }

    return ret as ConfigParams
```